src/Envelope.js

```diff
@@ -0,0 +1,107 @@
+import { Observable } from './Observe'
+import { knobToSeconds } from './utils/maths'
+
+class Envelope extends Observable {
+    constructor(context, audioParam) {
+      super()
+
+      this.context = context
+      this.audioParam = audioParam
+      // the following should work, but we need to convert that scientific notation to a value somehow
```

Comment:
https://stackoverflow.com/questions/10943997/how-to-convert-a-string-containing-scientific-notation-to-correct-javascript-num

Reply:
After thinking about it, we actually don't want to use the max and min values of the AudioNode, because for instance a gain node's maximum is extremely high and unusable. It should be defined manually by whatever is implementing it.